### ISSUE TITLE
File > New and ContextMenu New are inconsitent

### ISSUE THREAD

**Issue by laeubi** [eclipse-platform/eclipse.platform#306]
## If I choose `File` > `New` I get

![grafik](https://user-images.githubusercontent.com/1331477/210137502-508cc2bc-ec8e-4ebc-a10d-ab88618d4826.png)

## If I right click on a project in the plugin perspective I get

![grafik](https://user-images.githubusercontent.com/1331477/210137512-669c7cc8-91a1-4a55-bcc5-a0b1af4e0dd5.png)

## Expected
As a user I would expect that both menus are filled with similar content, at least a user has no clue here why these things are different. Also the menu with more items seems using a random order.

**Comment by Bananeweizen** [comment]
One of them is a context menu, and context menus react to selection, and context menus are space limited. So they should not really be the same. See the new shiny https://eclipse-platform.github.io/ui-best-practices/#context_menu for more reasoning :) I agree that _some_ more items would still be okay here.
The sort order is generally a very hard issue in systems where extensions are not managed centrally, since you don't really know what they are adding there, and what is more important to the user. The menu manager only knows that it gets a number of items from the extensions. That's why in several places of Eclipse this depends on the bundle names or similar "random" things. The platform as such only dictates the groups, to have at least some sorting. And just sorting every menu item group alphabetically would also not be nice, since a single plugin typically already sorts _its_ contributions in a meaningful way.

**Comment by mickaelistria** [comment]
> As a user I would expect that both menus are filled with similar content

I agree for the case of a new project (which is actually a global case and not a contextual one as the selection doesn't affect the proposal).

**Comment by laeubi** [comment]
Actually the selection do not influence anything if I choose `New Project ...` or `Others...` even worse, most (if not all) items offered from the "non contextual menu" does require a project where the item is created e.g. for sure I can save a product "everywhere" but I'm not a ware that the wizard allows me to chose any location (it always asks for a project location) so why should a project context not be sufficient?
